### ISSUE TITLE
Additional data structure

### ISSUE THREAD

**Issue by Lewenhaupt** [ncform/ncform#154]
Hi, 

I'm looking into how I could possible add an additional datastructure into ncform. My usecase is that I have a need to add "links" to dynamic values for certain fields. I must be able to persist this on my server and it should not be part of my schema. The links are not urls etc. but our own internal linking convention. Is my only option for this to fork the repo? I've tried looking but as runtime mixins aren't allowed in vue I can't find any obvious solution.

**Comment by daniel-dx** [comment]
Your question is whether the schema can change dynamically? If it is, just change the schema（new one） at runtime and re-assign it to ncform

**Comment by Lewenhaupt** [comment]
Not exactly. I would want a third object. Currently we have the formSchema and formValue. And I would like to introduce a third one, for example "linkStructure". I guess I could work around this by simply storing this in a vuex store or similar and commit to that object from a custom theme component.

**Comment by Lewenhaupt** [comment]
Hmm ok I looked into this a bit more. I can always access the closes FormItem fairly easily which gives me access to the formName which I can use to uniquely store an extra data structure somewhere else (such as a vuex store or similar). However, I can't seem to find a way to identify exactly where in the tree I currently am. "paths" to me sounds as if it should be, but it doesn't appear to give me the current location (only parent) and I've found, what I think is a bug with nested arrays.
Is "paths" the correct thing to be looking at here or should I gather this data from somewhere else?

**Comment by daniel-dx** [comment]
can you give me a online demo, and show your needs? i am not sure catch your meening

**Comment by Lewenhaupt** [comment]
No online demo available. But should the "paths" variable be a valida JMES path for the current field in the form?
